modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/TypeScriptNodeClientCodegen.java

```diff
@@ -96,22 +96,37 @@ public String toApiFilename(String name) {
         if (name.length() == 0) {
             return "default" + apiSuffix;
         }
+        if (importMapping.containsKey(name)) {
+            return importMapping.get(name);
+        }
```

Comment:
There are 4 methods here that convert a type name into some sort of file path. My first pass at this only updated one; this latest commit updates all of them. 

That this condition is checked in 4 different places is evidence that this could be addressed in a more general way. The `DefaultGenerator` already [checks the importMapping](https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultGenerator.java#L1152), but it's geared for imports that don't have file path dependencies (e.g. Java). In the TypeScript implementations, we need to construct the import differently based on whether it is mapped (in which case we use the mapped reference verbatim) or unmapped (in which case we attempt to construct a relative file path).